compiler/rustc_const_eval/src/const_eval/eval_queries.rs

```diff
@@ -336,13 +336,11 @@ pub fn eval_to_allocation_raw_provider<'tcx>(
 
             let alloc_id = mplace.ptr().provenance.unwrap();
 
-            // Validation failed, report an error.
-            if let Err(error) = validation {
-                Err(const_report_error(&ecx, error, alloc_id))
-            } else {
+            validation
+                // Validation failed, report an error.
+                .map_err(|error| const_report_error(&ecx, error, alloc_id))
                 // Convert to raw constant
-                Ok(ConstAlloc { alloc_id, ty: mplace.layout.ty })
-            }
+                .map(|()| ConstAlloc { alloc_id, ty: mplace.layout.ty })
```

Comment:
Honestly I find the `match` more readable than using higher-order functions... but I don't care very strongly.

Reply:
I force pushed away the commit that did this